### PR TITLE
ZK-4600: readonly spinner buttons should not be clickable

### PR DESCRIPTION
**Pull request by b952426**
ZK-4600: readonly spinner buttons should not be clickable

### CODE REVIEW

File: zktest/src/archive/test2/B96-ZK-4600.zul [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		Tue Mar 16 09:51:56 CST 2021, Created by katherinelin
+
+Copyright (C) 2021 Potix Corporation. All Rights Reserved.
+-->
+<zk>
+	<label>Click the buttons of readonly spinner, spinner value will not be changed.</label>
+	<div>readonly</div>
+	<spinner readonly="true" value="0" />
+	<doublespinner readonly="true" value="0" />
+    <div>disabled</div>
```
Comment (scribetw) [inline zktest/src/archive/test2/B96-ZK-4600.zul:19]: can be removed

File: zul/src/archive/web/js/zul/inp/less/combo.less [hunk 2]
```diff
@@ -130,6 +130,16 @@ each(@components, {
...
+			cursor: default;
+			&:hover{
+				border-color: @inputBorderColor;
+			}
+		}
+		&-readonly &-icon:hover {
+			background-color: @inputBackgroundColor;
+		}
+		&-readonly &-separator {
+			border-color: @inputBackgroundColor !important;
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/inp/less/combo.less:142]: avoid important
Ask the designer about the readonly style

**Review by scribetw — CHANGES_REQUESTED** [review]

File: zul/src/archive/web/js/zul/inp/less/combo.less [hunk 3]
```diff
@@ -91,13 +91,15 @@ each(@components, {
...
-		&-disabled {
+		&-disabled, &-readonly {
 			.opacity(@disabledOpacity);
 
 			& > input {
-				color: @inputDisableColor !important;
 				background: @inputDisableBackgroundColor !important;
 				cursor: default !important;
+				&:hover {
+					border-color: @inputBorderColor;
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/inp/less/combo.less:101]: readonly only

File: zul/src/archive/web/js/zul/inp/less/combo.less [hunk 4]
```diff
@@ -125,6 +130,19 @@ each(@components, {
...
+				&:focus {
+					border-color: @inputBorderColor;
+					& + .z-spinner-button, & + .z-doublespinner-button {
+						border-color: @inputBorderColor;
+					}
+				}
+				&:hover {
+					border-color: @inputBorderColor;
+					cursor: default;
+					& + .z-spinner-button, & + .z-doublespinner-button {
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/inp/less/combo.less:142]: Avoid using `:hover + .xxx` (see ZK-4153)

File: zul/src/archive/web/js/zul/inp/less/combo.less [hunk 5]
```diff
@@ -201,7 +219,7 @@ each(@components, {
 	& > i {
 		z-index: 2; // greater than arrow icon
 	}
-	&:hover > i {
+	&:not(&-readonly):hover > i {
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/inp/less/combo.less:222]:
```
&-readonly:hover > i {
    display: none;
}
```

**Review by scribetw — CHANGES_REQUESTED** [review]
don't modify zklessChecker
other themes

File: zul/src/archive/web/js/zul/inp/InputWidget.js [hunk 6]
```diff
@@ -507,6 +508,7 @@ zul.inp.InputWidget = zk.$extends(zul.Widget, {
 				window.scrollTo(windowX, windowY);
 		}
 		this._lastKeyDown = null;
+		if (this._readonly) jq(this.getInputNode()).removeClass(this.$s('readonly-focus'));
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/inp/InputWidget.js:511]: remove if
this.$s('focus')

File: zul/src/archive/web/js/zul/inp/less/combo.less [hunk 7]
```diff
@@ -48,6 +48,9 @@ each(@components, {
 		&-input&-hover, &-input&-hover + &-button {
 			border-color: @inputHoverBorderColor;
 		}
+		&-input&-readonly-focus, &-input&-readonly-focus + &-button {
```
Comment (DevChu) [inline zul/src/archive/web/js/zul/inp/less/combo.less:51]:
```
&-readonly .z-xxx-focus {
}
```

**Review by DevChu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**